commit 94b3de0b54053893ff1bd157de737cf9b113ddda
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-05-21T16:34:06+00:00 (5 months ago)
    
    TRITON-1676 linux-prepare-image should remove fstab entries added by cloud-init
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/tools/prepare-image/linux-prepare-image b/tools/prepare-image/linux-prepare-image
index 6d1e91a..4c5da8a 100755
--- a/tools/prepare-image/linux-prepare-image
+++ b/tools/prepare-image/linux-prepare-image
@@ -103,8 +103,9 @@ function cleanup_ssh() {
 }
 
 function cleanup_disks() {
-    echo "removing /dev/vdb entries from fstab"
-    sed -i '/^\/dev\/vdb/d' /etc/fstab
+    echo "removing /dev/vdb and cloud-init entries from fstab"
+    sed -i -e '/^\/dev\/vdb/d' -e '/[\t|,]comment=cloudconfig[,|\t]/d' \
+        /etc/fstab
 }
 
 function cleanup_metadata() {
